server.js

```diff
@@ -22,9 +17,80 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+// --- ALL THE ROUTES ---
+
+// --- front page ---
+app.get('/', (request /* incoming */, response /* outgoing */) => {
+
+  response.send('Books API.')
+})
+
+
+// --- return ALL BOOKS & search by TITLE using QUERY SELECTORS ---
+//localhost:8080/books?title=freakonomics
+
+app.get("/books", (request, response) => {
+
+  const titleSearch = request.query.title
+
+  if (titleSearch === undefined) {
+    response.json(booksData)
+  }
```

Comment:
Hmm, when would the titleSearch be `undefined`?

Reply:
I think if the title would have no entries i.e. the query would be empty. Like that: `/books?title `or `/books?title=` Correct me if I'm wrong.